### DISCUSSION TITLE
ANSI color for logs

### DISCUSSION THREAD

**Discussion by ejunker** [spatie/ray#74] [Ideas]
I use ANSI colors for my some of my logs. The logs show up in Ray but it shows the ANSI color codes instead of showing the text in different colors.

Example log entry with ANSI color codes:
```
[32m [1mHTTPS [0m [33m [1mGET [0m [33m [1m/debug [0m
```

**Comment by freekmurze** [comment]
Thanks for opening this discussion.

This currently is not supported. We simply just copy what gets writing to the log as a string to Ray.

If this can be solved in a simple way that works for you, I'd accept a PR.

**Comment by ejunker** [comment]
I was able to make a macro that seems to work

```php
Ray::macro('ansiLog', function ($ansi) {
    $html = (new \SensioLabs\AnsiConverter\AnsiToHtmlConverter())->convert($ansi);
    $this->send($html);
});
```

And then here is an example of how to use it:

```php
$color = new \Symfony\Component\Console\Color('blue', 'white');
ray()->ansiLog($color->apply('blue text'));
```

I'll spend more time using the macro and thinking about how this could work as a PR. Feel free to add this feature yourself if you want.

**Reply by freekmurze** [reply]
This looks promising, feel free to send a PR that adds this, and I'll review it.
